src/firefly/js/visualize/PlotViewUtil.js

```diff
@@ -142,36 +143,27 @@ export function hasWCSProjection(ref) {
  */
 export function getPlotViewIdList(visRoot, pvOrId, onlyIfPositionLocked=true, hasPlots=false) {
     if (!pvOrId) return [];
-    const pv= (typeof pvOrId ==='string') ? getPlotViewById(visRoot,pvOrId) : pvOrId;
-    const locked= visRoot.positionLock;
-    if (!locked && onlyIfPositionLocked) return [pv.plotId];
-    const idList=  visRoot.plotViewAry.map( (pv) => pv.plotId);
-    if (!hasPlots) return idList;
+    const pv = (typeof pvOrId === 'string') ? getPlotViewById(visRoot, pvOrId) : pvOrId;
 
-    return idList.filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
-}
+    const gid = pv.plotGroupId;
+    const group = getPlotGroupById(visRoot, gid);
 
-/**
- * Return an array of plotId's that are in the plot group associated with the the pvOrId parameter.
- * @param {VisRoot} visRoot - root of the visualization object in store
- * @param pvOrId this parameter will take the plotId string or a plotView object
- * @param onlyIfGroupLocked
- * @param hasPlots
- * @returns {Array.<String>}
- */
-export function getPlotViewIdListInOverlayGroup(visRoot,pvOrId,onlyIfGroupLocked=true, hasPlots=false) {
-    if (!pvOrId) return [];
-    const pv= (typeof pvOrId ==='string') ? getPlotViewById(visRoot,pvOrId) : pvOrId;
-    const gid= pv.plotGroupId;
-    const group= getPlotGroupById(visRoot,gid);
-    const locked= hasOverlayColorLock(pv,group);
-    if (!locked && onlyIfGroupLocked) return [pv.plotId];
-    const idList=  visRoot.plotViewAry.filter( (pv) => pv.plotGroupId===gid).map( (pv) => pv.plotId);
+    if (hasPlots) {
+        const targetLocked = visRoot.positionLock;
+        if (!targetLocked && onlyIfPositionLocked) return [pv.plotId];
+    }
+    else {
+      const overlayLocked = hasOverlayColorLock(pv, group);
+      if (onlyIfPositionLocked && !overlayLocked) return [pv.plotId];
+   }
+
+    const idList=  visRoot.plotViewAry.map( (pv) => pv.plotId);
     if (!hasPlots) return idList;
 
     return idList.filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
 }
 
+
```

Comment:
> `getPlotViewIdList` is several issues. You have fixed some but not all.
> 
> * I split it into two function `getPlotViewIdList`  and `getPlotViewIdListInOverlayGroup` when I revamped the wcs matching.  However, i did not change anything to call `getPlotViewIdListInOverlayGroup`.
> * Both functionality is still necessary but I think you correctly saw that most of the calls are _most_ concerned about overlays and color.
> * `getPlotViewIdList` does not need the `onlyIfPositionLocked` parameter anymore. No one is using it. I think you should remove it. Also removed `hasPlots` because it will be unnecessary it you do the next step.
> * The call in `HandlePlotCreation.js`, line 208, expects the original functionality. plot grouping is not an issue here, but `positionLock` is.  I took the old call and simplified it to work for that call. _see below._
> 
> ```js
> export function getPlotViewIdListByPositionLock(visRoot, pvOrId) {
>     if (!pvOrId) return [];
>     const pv= isString(pvOrId) ? getPlotViewById(visRoot,pvOrId) : pvOrId;
>     if (!visRoot.positionLock) return [pv.plotId];
>     return visRoot.plotViewAry
>         .map( (pv) => pv.plotId)
>         .filter( (id) => get(getPlotViewById(visRoot,id),'plots.length') );
> }
> ```

Yes, I noticed "onlyIfPositionLocked" and hasPlot seem useless but I did not want to make more changes because I am worried more bugs I may introduce.  OK, I will make the changes you suggested.